pkg/common-controller/snapshot_create_test.go

```diff
@@ -63,6 +63,8 @@ func TestCreateSnapshotSync(t *testing.T) {
 				map[string]string{
 					"snapshot.storage.kubernetes.io/deletion-secret-name":      "secret",
 					"snapshot.storage.kubernetes.io/deletion-secret-namespace": "default",
+					"csi.storage.k8s.io/snapshotter-secret-name":               "secret",
+					"csi.storage.k8s.io/snapshotter-secret-namespace":          "default",
```

Comment:
Why do you need to add this? Deletion secret is already added to content as annotation above.

Reply:
Please see my comment [here](https://github.com/kubernetes/enhancements/pull/2849#issuecomment-897759728).
This is to save the secret template itself as annotation without being resolved, not the resolved value, to be latter used to resolve again with different namespace or different volumeSnapshotContent name.
(For above comment case, it is used for just checking if the secret is defined through the template to see if transferring the snapshot could cause issues by secret).

Reply:
Also, check the discussion from 35:00. It was talking about PV, but the same thing is true to volumeSnapshotContent.
 https://youtu.be/hQZDcJmAB80?t=2105

Reply:
I'm not sure if it is safe to adjust the secret based on template like that.  I'd prefer a more explicit way of transferring secret. Here's an example: https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/docs/secret-export.md

Reply:
> Here's an example: https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/docs/secret-export.md

I agree with you that it would be good to provide a feature to move/copy the secret itself across the namespace by such a way (and it will be one of the another use cases for `StorageTransferRequest` and `StorageTransferApproval`).

However, what I'm discussing here is different from the use case. It is to update the reference to the secret that is defined in the VolumeSnapshotContent. After transfer, VolumeSnapshotContent may still referencing to the secret that may be only allowed to be accessed from the original namespace, as I explained [here]( https://github.com/kubernetes/enhancements/pull/2849#discussion_r692459041).
Therefore, the reference may need to be updated in some cases. And to do so, this feature will be of help (I'm not saying this is the only way and we can discuss if there will be a better way).